app/managers/performance_metrics_manager.ts

```diff
@@ -0,0 +1,41 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {getTimeSinceStartup} from 'react-native-startup-time';
+
+type Target = 'HOME' | 'CHANNEL' | 'THREAD' | undefined;
+type MetricName = 'channelSwitch' |
+    'teamSwitch';
+
+class PerformanceMetricsManager {
+    private startTimes: {[metricName: string]: undefined | number} = {};
+    private target: Target;
+
+    public setLoadTarget(target: Target) {
+        this.target = target;
+    }
+
+    public finishLoad(location: Target) {
+        if (this.target !== location) {
+            return;
+        }
+
+        getTimeSinceStartup().then((v) => console.log(`Took ${v}`));
+    }
+
+    public startMetric(metricName: MetricName) {
+        this.startTimes[metricName] = Date.now();
```

Comment:
On the web at least, `Date.now` isn't as precise as we might want it to be because of anti-fingeringprinting logic. Do we need to worry about that on mobile at all?

I'm also wondering if timezone changes or DST could have any effect on it since it's not guaranteed to always be increasing

Reply:
Looking deeper into this, seems like a `global.performance` object exist, with all the needed stuff (including performance.now()). I will try to migrate to that.

Reply:
Elsewhere, that API is part of the browser. I think Node.js only added their own version of it fairly recently, but it's as an imported package. I don't know if RN would have an equivalent to it on the JS side though